### PR TITLE
[DDW-555] Improve URL validation on Stake Pools

### PR DESCRIPTION
**Pull request by daniloprates**
This PR improves the error messages in the custom SMASH server url input.

## Todos

- [x] Improve the error messages
- [x] Copy & translation

## Screenshots

Screen Shot 2021-02-12 at 15 09 23
Screen Shot 2021-02-12 at 15 09 39
Screen Shot 2021-02-12 at 15 09 50
Screen Shot 2021-02-12 at 15 09 58

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---
## Test scenarios

**Scenario 1 - Insert URL without "https://"**
```gherkin
Given that I am on Stake Pools screen on "Settings"
And the option "Custom server" is selected
When I insert an URL that doesn't start with "https://"
Then the input is outlined in red
And the following message is displayed on a tooltip

- ENGLISH: The URL should start with "https://"
- JAPANESE: URLは「https://」から始めてください
```

**Scenario 2 - Insert URL with "https://" and a path**

```gherkin
Given that I am on Stake Pools screen on "Settings"
And the option "Custom server" is selected
When I insert an URL that start with "https://" and has a path 
Then the input is outlined in red
And the following message is displayed on a tooltip

- ENGLISH: Only "https://" protocol and hostname (e.g. domain.com) are allowed
- JAPANESE:「https://」プロトコルおよびホスト名（例：domain.com）のみ使用できます
```

**Scenario 3 - Insert URL with "https://" and no path**

```gherkin
Given that I am on Stake Pools screen on "Settings"
And the option "Custom server" is selected
When I insert an URL that start with "https://"
Then no error is displayed on the input
```

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes
- [ ] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [ ] In English
  - [ ] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/settings/categories/StakePoolsSettings.js [hunk 1]
```diff
@@ -192,6 +205,16 @@ export default class StakePoolsSettings extends Component<Props, State> {
 
   handleIsValid = (url: string) => url === '' || SMASH_URL_VALIDATOR.test(url);
 
+  handleErrorMessage = (value: string) => {
+    const { intl } = this.context;
+    let errorMessage = messages.invalidUrl;
+    const { pathname, search } = getUrlParts(value);
+    if (!/^https:\/\//i.test(value)) errorMessage = messages.invalidUrlPrefix;
```
Comment (DominikGuzei) [inline source/renderer/app/components/settings/categories/StakePoolsSettings.js:212]: Just a small thing but I wonder if that `if/else` statement (without curly braces) passes linting?

**Review by DominikGuzei — APPROVED** [review]
Code looks good IMO 👍

File: source/renderer/app/components/settings/categories/StakePoolsSettings.js (hunk 1 above)
Comment (daniloprates) [reply source/renderer/app/components/settings/categories/StakePoolsSettings.js:212]: @DominikGuzei it does :)

**Comment by gnpf** [comment]
@daniloprates Looks good, just one detail:

- URLs with no extension are considered valid (no error is displayed before submitting it). Could this be improved? Check [this video](https://drive.google.com/file/d/1n2XISV6HAdtZ0IHFkKEqWP4T0m15LWAR/view?usp=sharing).

I added some test scenarios on this PR.

File: source/renderer/app/components/settings/categories/StakePoolsSettings.js (hunk 1 above)
Comment (DominikGuzei) [reply source/renderer/app/components/settings/categories/StakePoolsSettings.js:212]: interesting, maybe we should adjust our linting rules 😆

**Comment by daniloprates** [comment]
> @daniloprates Looks good, just one detail:
> 
> * URLs with no extension are considered valid (no error is displayed before submitting it). Could this be improved? Check [this video](https://drive.google.com/file/d/1n2XISV6HAdtZ0IHFkKEqWP4T0m15LWAR/view?usp=sharing).
> 
> I added some test scenarios on this PR.

@gnpf we can't prevent this. Because of new top-level domains, a URL can be valid without the extension.

File: source/renderer/app/components/settings/categories/StakePoolsSettings.js (hunk 1 above)
Comment (daniloprates) [reply source/renderer/app/components/settings/categories/StakePoolsSettings.js:212]: 😅 I find it much easier to visualize simple checks like these without braces, but can add them :)

**Comment by a-rukin** [comment]
image
shouldn't we hide this when we focus other field?

**Comment by ManusMcCole** [comment]
Hi @a-rukin.It seems to be hiding ok for me on the Mac builds  16487  <a href="https://drive.google.com/file/d/1i1ifvQRXuVB0icEr-ywK6RR_nD8x3y_4/view?usp=sharing">video</a>. What Os did you test on ?

**Comment by a-rukin** [comment]
> Hi @a-rukin .It seems to be hiding ok for me on the Mac builds 16487 [video](https://drive.google.com/file/d/1i1ifvQRXuVB0icEr-ywK6RR_nD8x3y_4/view?usp=sharing). What Os did you test on ?

Just built latest commit 🤷

**Review by ManusMcCole — APPROVED** [review]
👍

**Comment by a-rukin** [comment]
image
Pls use "font-medium" here instead of "font-bold"

**Review by a-rukin — APPROVED** [review]
Looks good

**Review by nikolaglumac — APPROVED** [review]
Great work @daniloprates - now this looks 💅

**Pull request merged**